### ISSUE TITLE
flatpak manifest

### ISSUE THREAD

**Issue by gthreepw00d** [enz/pentobi#10]
```yaml
app-id: io.sourceforge.pentobi
runtime: org.kde.Platform
runtime-version: '5.15'
sdk: org.kde.Sdk
command: /app/bin/pentobi
cleanup:
  - '/lib/debug'
finish-args:
  - --socket=x11
  - --socket=wayland
  - --share=ipc
  - --device=dri
modules:
  - name: pentobi
    buildsystem: cmake
    config-opts:
      - -DCMAKE_BUILD_TYPE=Release
      - -DDOCBOOKXSL_DIR=/usr/share/xml/docbook/xml/xsl-stylesheets
#no qt5-qtwebview in sdk
      - -DPENTOBI_OPEN_HELP_EXTERNALLY=ON
    sources:
      - type: git
        url: https://github.com/enz/pentobi.git
        tag: v18.3
```

**Comment by enz** [comment]
Could you say a few words what this issue is about? Are you proposing that I create flatpak images of Pentobi for download?

Generally I don't want to become involved with creating binary packages of Pentobi. Packaging is a nice way for external contributors to help out and can be done in separate projects.

**Comment by gthreepw00d** [comment]
I don't know, sharing my work. Though flatpak package would be nice.

**Comment by enz** [comment]
I see. Thanks for sharing. Then I will leave the issue open for a while in case anyone wants to pick it up. But that would have to be moved to an external project then.

Myself I am a big fan of Linux distributions where binaries are created from the sources by a trusted distribution and you don't have to install binaries directly from dozens of unknown and potentially dangerous sources. It is also a good test for me how much interest there is in Pentobi. If enough people are interested in Pentobi, someone will package it. If not, it is probably not worth packaging.

On a side note: I am not familiar with flatpack, but the KDE SDK seems like a big dependency. You need only Qt for Pentobi, unless you want to compile its thumbnailer plugin for KDE, which is optional. This is why there are two packages on Debian: pentobi and pentobi-kde-thumbnailer.

**Comment by Mailaender** [comment]
I submitted something similar to https://github.com/flathub/flathub/pull/3149.

**Comment by enz** [comment]
[copying this comment, which I accidentally added to the flathub issue]

Great. If you are interested in maintaining the flatpack config, I could add a link from the Pentobi homepage.

Also note that PENTOBI_OPEN_HELP_EXTERNALLY is no longer needed with Pentobi >= 19.0 because it always uses an external browser (and it won't be needed in the future because Pentobi >= 21.0 will include a custom help viewer impementation, which neither needs QtWebView nor an external browser).

**Comment by enz** [comment]
Just for the record: Now that I started using Flathub myself, I actually like the flatpak packaging system, mainly because of the extra security of running apps in sandboxes. And with the first distro (Fedora) switching to flatpak, it seems to be the future of packaging on Linux. I added a verification token on pentobi.sourceforge.io for Flathub and will test coming Pentobi releases for compatibility with flatpak.
